### PR TITLE
Add Linux_Sources package and recreate linuxheaders package based on it

### PR DESCRIPTION
**Pull request by JL2210**
@uberhacker Binaries, please.

I hope this makes you all happy :).

### CODE REVIEW

File: packages/linuxheaders.rb [hunk 1]
```diff
@@ -2,18 +2,24 @@
...
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew-cross/linuxheaders-3.18-2-chromeos-x86_64.tar.xz',
-  })
-  binary_sha256 ({
-    aarch64: '811e1bfa4d83e866ac7625c6970d7289b8768bf30b18af8a3b79c0760dfa42af',
-     armv7l: '811e1bfa4d83e866ac7625c6970d7289b8768bf30b18af8a3b79c0760dfa42af',
-       i686: '8ee5c68467cb5e7994993cddf163d59cb61a5f7d70f5bc595cb3fe288131ae59',
-     x86_64: '830b18e54ebd6a2c8701c5ea1efd883e345719770de34ad0ece2f7cfb904df6f',
-  })
+  homepage 'https://kernel.org/'
+  version '3.18-1'
```
Comment (uberhacker) [inline packages/linuxheaders.rb]: This really should be `version '3.18-3'`.

Comment (JL2210) [reply packages/linuxheaders.rb]: Oh, shoot. I meant to change that back to `3.18` after testing. I imagine by now that that version of the package is so ungodly old that nobody will have it installed.

**Comment by JL2210** [comment]
I'm making some progress on Musl support. Once I gave up on multilib, the only issue stopping me now is disk space. It's unhelpful that Gentoo Prefix takes up so much space.

**Comment by JL2210** [comment]
After this is merged, we should be able to delete https://bintray.com/chromebrew/chromebrew-cross

**Comment by JL2210** [comment]
@uberhacker?

File: packages/linux_sources.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  source_url 'https://www.kernel.org/pub/linux/kernel/v3.x/linux-3.18.tar.xz'
+  source_sha256 'becc413cc9e6d7f5cc52a3ce66d65c3725bc1d1cc1001f4ce6c32b69eb188cbd'
+
+  def self.install
+    case ARCH
+    when 'x86_64', 'x86'
+      arch = 'x86'
+    when 'armv7l', 'aarch64'
+      arch = 'arm'
+    end
```
Comment (uberhacker) [inline packages/linux_sources.rb]: Why do we define the `arch` variable and never use it?  Unless I'm missing something here?

Comment (JL2210) [reply packages/linux_sources.rb]: Sorry. Little mistake.

Comment (JL2210) [reply packages/linux_sources.rb]: Done.

**Comment by JL2210** [comment]
I'm going to add a quick patch to this that should fix the Sommelier build process. Hang on.

**Comment by JL2210** [comment]
Please build `linuxheaders` and this will be good. The patches that Chrome OS uses for the linuxheaders are way too invasive.

**Comment by uberhacker** [comment]
What do you mean by invasive?  Can you be more specific?

**Comment by JL2210** [comment]
The patches that Chromium applies to the linux sources are too invasive and would require major changes to the programs and libraries that use them.

**Comment by uberhacker** [comment]
You are not being specific enough.  It what ways specifically are they being invasive?  We currently use this supposedly "invasive" linuxheaders package without any major changes.

**Comment by JL2210** [comment]
Dude, read. I'm deciding against applying the Chromium patches for the Linux headers because the patches are invasive and will break stuff, not the headers. Build binaries and this is good.

**Comment by uberhacker** [comment]
Notice during build:
```
rm: cannot remove '.': Is a directory
```
Probably should find a more elegant solution.

**Comment by JL2210** [comment]
Does that make the build fail? It shouldn't affect anything.

**Comment by uberhacker** [comment]
No but do you know how to fix to remove the notice?

**Comment by JL2210** [comment]
Append `-not -type d` to the `find .` command.

**Comment by JL2210** [comment]
Done.

**Comment by JL2210** [comment]
@uberhacker? I've done all that you told me to.

**Comment by uberhacker** [comment]
I held back on this because I want to do more testing.

**Comment by JL2210** [comment]
Well, it works fine on my end. Any issues with the other architectures? I'd be happy to fix.

**Comment by JL2210** [comment]
@uberhacker Just so you know, I've been using this since forever ago and it works fine. Do you mind building binaries for `linuxheaders`?

**Pull request merged**